static/js/stream_edit.js

```diff
@@ -500,7 +500,7 @@ function change_stream_privacy(e) {
         history_public_to_subscribers: JSON.stringify(history_public_to_subscribers),
     };
 
-    if (page_params.is_owner) {
+    if (page_params.zulip_plan_is_not_limited && page_params.is_owner) {
```

Comment:
I think the bug here is actually that we're sending this value when it hasn't been changed by the user, not an issue with the plan type.  We should only be including it when `message_retention_days` is different from the `sub.` value, right?

@pragatiagrawal31 @shubhamdhama @ryanreh99 FYI since there's other work on this modal.

Reply:
Okay. We do this for other settings in the modal also, so should I also change those to send only when the value has been changed

Reply:
Yes, the correct way that all of our settings UI widgets should work is to only send values where things have changed.  The stream name might be required to identify which stream was modified; if not we should indeed remove it.

(Yeah, I understand why the bug only affects limited plans right now, but we should fix the structural issue here)